### PR TITLE
#2001 made notes optional

### PR DESCRIPTION
**Pull request by superhvarn**
## Changes

- made migration after editing schema to make notes optional

 - made some backend changes with material classes and projects route
 - made some frontend changes with the text to display "notes (optional)".
Screenshot 2024-02-05 at 7 34 58 PM

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #2001 )

### CODE REVIEW

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/BOM/MaterialForm/MaterialFormView.tsx [hunk 1]
```diff
@@ -261,7 +261,7 @@ const MaterialFormView: React.FC<MaterialFormViewProps> = ({
         </Grid>
         <Grid item xs={12}>
           <FormControl fullWidth>
-            <FormLabel>Notes</FormLabel>
+            <FormLabel>Notes (Optional)</FormLabel>
```
Comment (Aaryan1203) [inline src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/BOM/MaterialForm/MaterialFormView.tsx:264]: make it a lowercase 'o' just so its consistent with the others

**Review by Aaryan1203 — CHANGES_REQUESTED** [review]
Your typescript tests are still failing. i will take a look later today at how we can fix that

**Review by Aaryan1203 — CHANGES_REQUESTED** [review]
When i tested it right now, i am still getting an error when submitting without putting something for notes saying that the notes are a required field. I would say check the frontend checks to make sure its not still expecting one

File: src/frontend/src/utils/bom.utils.ts [hunk 2]
```diff
@@ -32,7 +32,7 @@ export const materialToRow = (material: Material, idx: number): BomRow => {
     price: `$${centsToDollar(material.price)}`,
     subtotal: `$${centsToDollar(material.subtotal)}`,
     link: material.linkUrl,
-    notes: material.notes
+    notes: material.notes ?? 'None'
```
Comment (Aaryan1203) [inline src/frontend/src/utils/bom.utils.ts:35]: don't think you need this

**Review by Aaryan1203 — APPROVED** [review]
Tested, looks good to me!

**Review by Aaryan1203 — APPROVED** [review]

**Review by RChandler234 — APPROVED** [review]
LGTM 🔥

**Pull request merged**